src/core/stream_send.c

```diff
@@ -159,7 +159,7 @@ QuicStreamSendShutdown(
             QUIC_STREAM_SEND_FLAG_FIN,
             DelaySend);
 
-    } else {
+    } else if (Stream->ReliableOffsetSend == 0) {
 
         //
```

Comment:
If `Stream->Flags.LocalCloseReset` is already set, then we've already sent a (non-reliable) reset, and there is no need to do anything more.